rfq/stream.go

```diff
@@ -0,0 +1,306 @@
+
+	// Send the quote request to the RFQ manager.
+	var msg rfqmsg.IncomingMsg = quoteRequest
+	sendSuccess := fn.SendOrQuit(h.cfg.IncomingMessages, msg, h.Quit)
+	if !sendSuccess {
+		return fmt.Errorf("RFQ stream handler shutting down")
+	}
+
+	return nil
+}
+
+// handleIncomingQuoteAccept handles an incoming quote accept peer message.
+func (h *StreamHandler) handleIncomingQuoteAccept(
+	wireMsg rfqmsg.WireMessage) error {
+
+	quoteAccept, err := rfqmsg.NewAcceptFromWireMsg(wireMsg)
+	if err != nil {
+		return fmt.Errorf("unable to create a quote accept message "+
+			"from a wire message: %w", err)
+	}
+
+	log.Debugf("Stream handling incoming message (msg_type=%T, "+
+		"origin_peer=%s)", quoteAccept, quoteAccept.MsgPeer())
+
+	// TODO(ffranr): At this point, we need to validate that the quote
+	//  accept message is valid. We need to check that the quote accept
+	//  message is associated with a valid quote request.
+
+	// Send the message to the RFQ manager.
+	var msg rfqmsg.IncomingMsg = quoteAccept
+	sendSuccess := fn.SendOrQuit(h.cfg.IncomingMessages, msg, h.Quit)
+	if !sendSuccess {
+		return fmt.Errorf("RFQ stream handler shutting down")
+	}
+
+	return nil
+}
+
+// handleIncomingQuoteReject handles an incoming quote reject peer message.
+func (h *StreamHandler) handleIncomingQuoteReject(
+	wireMsg rfqmsg.WireMessage) error {
+
+	quoteReject, err := rfqmsg.NewQuoteRejectFromWireMsg(wireMsg)
+	if err != nil {
+		return fmt.Errorf("unable to create a quote reject message "+
+			"from a wire message: %w", err)
+	}
+
+	log.Debugf("Stream handling incoming message (msg_type=%T, "+
+		"origin_peer=%s)", quoteReject, quoteReject.MsgPeer())
+
+	// TODO(ffranr): At this point, we need to validate that the quote
+	//  reject message is valid. We need to check that the quote reject
+	//  message is associated with a valid quote request.
+
+	// Send the message to the RFQ manager.
+	var msg rfqmsg.IncomingMsg = quoteReject
+	sendSuccess := fn.SendOrQuit(h.cfg.IncomingMessages, msg, h.Quit)
+	if !sendSuccess {
+		return fmt.Errorf("RFQ stream handler shutting down")
+	}
+
+	return nil
+}
+
+// handleIncomingWireMessage handles an incoming wire message.
+func (h *StreamHandler) handleIncomingWireMessage(
+	wireMsg rfqmsg.WireMessage) error {
+
+	switch wireMsg.MsgType {
+	case rfqmsg.MsgTypeRequest:
+		err := h.handleIncomingQuoteRequest(wireMsg)
+		if err != nil {
+			return fmt.Errorf("unable to handle incoming quote "+
+				"request message: %w", err)
+		}
+
+	case rfqmsg.MsgTypeAccept:
+		err := h.handleIncomingQuoteAccept(wireMsg)
+		if err != nil {
+			return fmt.Errorf("unable to handle incoming quote "+
+				"accept message: %w", err)
+		}
+
+	case rfqmsg.MsgTypeReject:
+		err := h.handleIncomingQuoteReject(wireMsg)
+		if err != nil {
+			return fmt.Errorf("unable to handle incoming quote "+
+				"reject message: %w", err)
+		}
+
+	default:
+		// Silently disregard the message if we don't recognise the
+		// message type.
+		return nil
```

Comment:
Should we at least log an info or debug line here? Or is this to avoid spamming if other custom wire messages are received? Maybe a debug line like "Received message %T which is unrelated to RFQ, ignoring" could still be helpful.

Reply:
I was trying to avoid scaling the number of log messages with the number of peer messages received by the LND node. Perhaps ee should add support in LND to be able to subscribe to certain message types only.

I've added a trace level log message. Let me know if you'd prefer debug level.